src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
```

Comment:
Think about which variables should be local in the method and which should be class-level. The 'GeometricFigureSupplier' and 'ColorSupplier' instances could be class-level variables to avoid creating new instances each time the method is called.